### DISCUSSION TITLE
v2.0.0-beta.1

### DISCUSSION THREAD

**Discussion by jameslittle230** [jameslittle230/stork#342] [Announcements]
This is the first beta release of the Stork 2.0.0 release branch. The beta signifier means that the release is **documented** and **usable**, though the API is not final and not all features have been implemented yet.

Until 2.0.0 is released, the source code is stored in the [v2 branch](https://github.com/jameslittle230/stork/tree/v2) of the repository.

The documentation is available [**here**](https://63e05aa8ce4d5300089b2b8e--stork-search.netlify.app).

---

2.0.0 greatly simplifies the search algorithm. As a result, the indexing binary, WASM blob, and Javascript file are all smaller than the 1.x branch. Searches are faster. The configuration API is more powerful. 

The following features are still not working:

* Sharded indexes. The file format supports them, the library supports merging shards together, but the builder doesn't yet know how to split an index into multiple parts.
* Stemming
* HTML parsing, including the included and excluded selectors
* Matching search results in titles
* Result and excerpt scoring, leading to strange ordering of results

Currently, only search indexes made with the 2.0.0 indexer binary can be used with the 2.0.0 web code. This lets me greatly simplify the support matrix, so I'm inclined to leave it that way; please let me know if this would be troublesome for your workflow.

**Please give me feedback if you have any!**

---

The following artifacts have been uploaded to the Stork CDN:

- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork.js`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork.wasm`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork.css`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork-ubuntu-20-04`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork-ubuntu-22-04`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork-macos-11`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/stork-macos-12`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/federalist.st`
- `https://files.stork-search.net/releases/v2.0.0-beta.1/3b1b.st`

<hr /><em>This discussion was created from the release <a href='https://github.com/jameslittle230/stork/releases/tag/v2.0.0-beta.1'>v2.0.0-beta.1</a>.</em>

**Comment by vchirikov** [comment]
Could you provide win build or publish beta build to crates.io ?
And seems like you missed .map file, isn't it?

**Comment by danj-ca** [comment]
@jameslittle230 Thanks for Stork! This is a great little search engine — I found it when I switched to Pelican as my static site generator of choice, because the official Pelican search plugin uses Stork.

A big gap for me right now is the lack of "exact" searches, conventionally done using quotes. For example, searching for `I fell off the roof` finds occurrences of any and all of those words.

If I want to search only for the exact phrase, I would expect to enter `"I fell off the roof"`, but this appears to search for literally the word 'if' prefixed with a leading quotation mark, and also for the word 'roof' followed by a trailing quotation mark, and also the word 'fell', etc. 😬

Are exact-phrase searches on the roadmap? Maybe for v2? 👀

**Reply by jameslittle230** [reply]
This is a good idea! Thanks for describing it so clearly. :)

There's a little bit of handling in the V2 beta track for exact results, but it's still a work-in-progress. It's also focused on _exact word_ matches instead of _exact phrase_ matches, but I agree that exact phrase matches seem like the more important behavior here.

I can't make any promises but I'll see if I can make some efforts here!

-James
